src/index.js

```diff
@@ -4,14 +4,15 @@ import { PluginSidebar } from '@wordpress/editor';
 import { PanelRow, PanelBody, Button, FlexItem, Flex } from '@wordpress/components';
 import { registerPlugin, unregisterPlugin } from '@wordpress/plugins';
 import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from '@wordpress/element';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 import { store as editPostStore } from '@wordpress/edit-post';
 import { useEntityRecords } from '@wordpress/core-data';
 import { Icon, caution } from '@wordpress/icons';
 
 import { ReactComponent as ScalesIcon } from './assets/scale-icon.svg';
 
-const { mediaThreshold, featuredImageSize } = window.mediaWeightData;
+const { mediaThreshold, featuredImageSize, previewPostLink } = window.mediaWeightData;
```

Comment:
You can get this dynamically in the JS from `const { getEditedPostPreviewLink } = useSelect( select => select('core/editor') )`. Because when you create a new post vs editing an existing I think there won't necessarily be an ID yet for previewing in the backend. I might be wrong but will be good to check. I would recommend doing it in JS at any rate to be safe.

https://developer.wordpress.org/block-editor/reference-guides/data/data-core-editor/#geteditedpostpreviewlink

Reply:
I updated the code to use `getEditedPostPreviewLink`.